pkg/importer/dataStream_test.go

```diff
@@ -7,416 +7,404 @@ import (
+	rdrs2 := ioutil.NopCloser(strings.NewReader("test data for reader 2"))
+	rdrA := reader{4, rdrs1}
+	rdrB := reader{7, rdrs2}
 
-	ds := &DataStream{
-		url:         dsurl,
-		buf:         make([]byte, image.MaxExpectedHdrSize),
-		accessKeyID: accKey,
-		secretKey:   secKey,
-	}
+	rdrsTest := []reader{rdrA, rdrB}
 
-	ds.constructReaders()
+	It("Should successfully close readers", func() {
+		err := closeReaders(rdrsTest)
+		Expect(err).NotTo(HaveOccurred())
+	})
+})
 
-	return ds
-}
+var _ = Describe("Random file name", func() {
+	const numbyte = 8
 
-func createDataStreamBytes(testfile, ep string, defaultBuf, singlereader bool) (*DataStream, []byte, error) {
-	urlPath := filepath.Join(ep, testfile)
-	if len(ep) == 0 {
-		urlPath = getURLPath(testfile)
-	}
-	testFilePath := getTestFilePath(testfile)
-	ds := createDataStream(urlPath, "", "")
+	It("create expected random name", func() {
+		randName := make([]byte, numbyte)
+		rand.Read(randName)
+		wantString := hex.EncodeToString(randName)
 
-	// if no file supplied just return empty byte stream
-	if len(testfile) == 0 && !defaultBuf {
-		return ds, nil, nil
-	}
-	if len(testfile) == 0 && defaultBuf {
-		return ds, []byte{'T', 'E', 'S', 'T'}, nil
-	}
+		got := randTmpName("testfile.img")
+		_, fn := filepath.Split(got)
 
-	f, _ := os.Open(testFilePath)
-	defer f.Close()
-	testBytes, err := ioutil.ReadAll(f)
-	if err != nil {
-		return nil, nil, fmt.Errorf("Unable to read datastream buffer")
-	}
+		Expect(len(fn)).To(Equal(len("testfile.img") + len(wantString)))
+	})
+})
 
-	return ds, testBytes, nil
-}
+var _ = Describe("Streaming Data Conversion", func() {
 
-func getFileSize(testfile string) (int, error) {
-	f, err := os.Open(getTestFilePath(testfile))
-	defer f.Close()
+	tinyCoreIso := "tinyCore.iso"
+	baseTestImage, err := filepath.Abs(filepath.Join(TestImagesDir, tinyCoreIso))
 	if err != nil {
-		return 0, fmt.Errorf("Unable to open source datastream file %s", getTestFilePath(testfile))
+		Fail(fmt.Sprintf("Error getting abs path: %v\n", err))
 	}
-	fi, err := f.Stat()
-	if err != nil {
-		return 0, fmt.Errorf("Unable to stat file %v", f.Name())
-	}
-
-	return int(fi.Size()), nil
-}
-
-func getTestFilePath(testfile string) string {
-	// CWD is set within go test to the directory of the test
-	// being executed, so using relative path
-	imageDir, _ := filepath.Abs(TestImagesDir)
-	return filepath.Join(imageDir, testfile)
-}
 
-func getURLPath(testfile string) string {
-	// CWD is set within go test to the directory of the test
-	// being executed, so using relative path
-	imageDir, _ := filepath.Abs(TestImagesDir)
-	return "file://" + imageDir + "/" + testfile
-}
+	var tmpTestDir string
+	BeforeEach(func() {
+		tmpTestDir = testDir(os.TempDir())
+		if err != nil {
+			Fail(fmt.Sprintf("[BeforeEach] Failed created test dir: %v\n", err))
+		}
+		By(fmt.Sprintf("[BeforeEach] Creating temporary dir %s", tmpTestDir))
+		syscall.Umask(0000)
```

Comment:
not sure this is needed. Were you getting perms errors in your testing?

Reply:
I copied this directly out of the original file, not sure why it is needed.